src/components/sections/judge/JudgeContests.tsx

```diff
@@ -0,0 +1,207 @@
+'use client'
+
+import {DateTime} from 'luxon'
+import {useMemo} from 'react'
+
+import JudgeContestsTable from './JudgeContestTable'
+
+import Skeleton from '@/components/ui/Skeleton'
+import {Tabs, TabsList, TabsTrigger} from '@/components/ui/Tabs'
+import {
+  mergeSearchParamsUpdaters,
+  useSearchParamsEnumArrayState,
+  useSearchParamsEnumState,
+  useSearchParamsNumericState,
+  useUpdateSearchParams,
+} from '@/lib/hooks/useSearchParamsState'
+import {formatTabCount} from '@/lib/utils/common/format'
+import Separator from '@/components/ui/Separator'
+import {ContestOccurence, ContestStatus} from '@/server/db/models'
+import TablePagination from '@/components/ui/TablePagination'
+import {JudgeContestSorting} from '@/lib/types/enums'
+import {useSortingSearchParams} from '@/lib/hooks/useSortingSearchParams'
+import {useGetJudgeContests} from '@/lib/queries/contest/getJudgeContests'
+import {Filter, FilterControls} from '@/components/ui/Filter'
+import {selectOptions} from '@/lib/utils/common/enums'
+
+export const JUDGE_CONTESTS_PAGE_SIZE = 10
+
+const JudgeContests = () => {
+  const updateSearchParams = useUpdateSearchParams()
+  const [contestType, {getSearchParamsUpdater: updateContestTypeSearchParams}] =
+    useSearchParamsEnumState('type', ContestOccurence, ContestOccurence.FUTURE)
+  const [page, {getSearchParamsUpdater: updatePageSearchParams}] =
+    useSearchParamsNumericState('page', 1)
+  const [sortParams, {getSortParamsUpdaters: updateSortSearchParams}] =
+    useSortingSearchParams(JudgeContestSorting)
+  const [contestStatus, {getSearchParamsUpdater: updateStatusSearchParams}] =
+    useSearchParamsEnumArrayState('status', ContestStatus)
+
+  const {data: contests, isLoading} = useGetJudgeContests({
+    type: contestType,
+    pageParams: {
+      limit: JUDGE_CONTESTS_PAGE_SIZE,
+      offset: (page - 1) * JUDGE_CONTESTS_PAGE_SIZE,
+    },
+    sort: sortParams,
+    status: contestStatus,
+  })
+
+  const liveContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) =>
+          DateTime.fromJSDate(contest.startDate) < DateTime.now() &&
+          DateTime.fromJSDate(contest.endDate) > DateTime.now(),
+      ),
+    [contests],
+  )
+
+  const futureContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) => DateTime.fromJSDate(contest.startDate) > DateTime.now(),
+      ),
+    [contests],
+  )
+
+  const pastContests = useMemo(
+    () =>
+      contests?.data.filter(
+        (contest) => DateTime.fromJSDate(contest.endDate) < DateTime.now(),
+      ),
+    [contests],
+  )
```

Comment:
Is this necessary? Isn't it already filtered on BE?

Reply:
You're right. Removed it for `MyFindings` as well